File: App04_News/app/src/main/java/com/callor/topnews/MainActivity.java
package com.callor.topnews;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.callor.topnews.databinding.ActivityMainBinding;
import com.callor.topnews.service.NaverService;
import com.callor.topnews.service.impl.NaverNewsServiceImplV1;

import java.text.SimpleDateFormat;
import java.time.LocalDate;
import java.util.Date;

public class MainActivity extends AppCompatActivity {

    /*
    activity.xml 파일에 선언된 view Component를 사용할때
    findByViewId()를 사용하지 않고 접근할 수 있도록 선언된 경우
        buildFeatures{
            viewBinding true
        }
    자동으로 activity_main.xml 파일이름을 확장하여
    ActivityMainBinding이라는 클래스가 생성된다
     */
    ActivityMainBinding main_binding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        /*
        기존에 사용하던 setContentView(R.layout.activity_main); 를

        main_binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(main_binding.getRoot());
        코드로 변경

        이 코드로 시작이 되면 activity.xml 파일에 있는
        모든 view Component가 한꺼번에 사용가능한 상태로 변경된다다
         */
       setContentView(R.layout.activity_main);

        main_binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(main_binding.getRoot());

        SimpleDateFormat sd = new SimpleDateFormat( "yyyy-MM-dd"  );

        String curDate = sd.format(new Date(System.currentTimeMillis()));

        NaverService naverService = new NaverNewsServiceImplV1(main_binding.newsListView);
        naverService.getNews("코기");


    }
}

File: Empty/app/src/main/java/com/example/empty/MainActivity.kt
package com.example.empty

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.example.empty.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

//    lateinit var : 지금 변수 선언만 하고 생성은 잠시 후에 실행하겠다
    private lateinit var mainBinding:ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        선언된 mainBinding 을 activity_main.xml 파일을 열어서 inflate(확장)
//        하여 mainBinding 객채로(변수로) 초기화하기
        mainBinding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(mainBinding.root)

        val navController = findNavController(R.id.fragmentContainer)
        val appBarConfig = AppBarConfiguration(
            setOf(
                R.id.bottom_nav_home, R.id.bottom_nav_book
            )
        )

        setupActionBarWithNavController(navController, appBarConfig)
        mainBinding.bottomNav.setupWithNavController(navController)
    }
}